### PR TITLE
Add ability to paste entities to any host type

### LINKED ISSUE #488: Clipboard.pasteEntities does not have ability to select domain/avatar/local
It would be nice if pasteEntities had the same options as addEntity to set entity type.

### PR DESCRIPTION
**Pull request by kasenvr**
It defaults to "domain" if not specified.

Closes #488 

Test plan, make sure that:
- you can still call the method without specifying host type
- it works for all three host types and that the functionality works as expected for all three host types.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8ce3ce3_d9e0605b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8ce3ce3.tar_d9e0605b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8ce3ce3_d9e0605b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8ce3ce3.tar_d9e0605b.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8ce3ce3_d9e0605b.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b.tar_d881e1e3.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b.tar_d881e1e3.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.exe

**Comment by 1st-BrainStormer** [comment]
It doesn't seem to work the way I would expect.  It should override any of the settings in the imported/clipped  JSON if they exist.    
i.e. : 
    "clientOnly": true,
    "avatarEntity": true,
    "localEntity": false

**Comment by 1st-BrainStormer** [comment]
On a side note, we really should also change "export entities" to use entityHostType instead of the old system going forward to avoid confusion.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -5545,8 +5545,8 @@ bool Application::importEntities(const QString& urlOrFilename, const bool isObse
     return success;
 }
 
-QVector<EntityItemID> Application::pasteEntities(float x, float y, float z) {
-    return _entityClipboard->sendEntities(&_entityEditSender, getEntities()->getTree(), x, y, z);
+QVector<EntityItemID> Application::pasteEntities(QString entityHostType, float x, float y, float z) {
```
Comment (HifiExperiments) [inline interface/src/Application.cpp:5548]: All these other ones should be const & too

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b.tar_d881e1e3.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b.tar_d881e1e3.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-193349b_d881e1e3.exe

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -5545,8 +5545,8 @@ bool Application::importEntities(const QString& urlOrFilename, const bool isObse
     return success;
 }
 
-QVector<EntityItemID> Application::pasteEntities(float x, float y, float z) {
-    return _entityClipboard->sendEntities(&_entityEditSender, getEntities()->getTree(), x, y, z);
+QVector<EntityItemID> Application::pasteEntities(const QString entityHostType, const float x, const float y, const float z) {
```
Comment (HifiExperiments) [inline interface/src/Application.cpp:5548]: const QString&, not just const QString

and actually it’s better to leave floats as just floats!  primitives and small objects are better left alone, the copy is actually less expensive than the conversion to a reference (which is actually a pointer).

File: libraries/entities/src/EntityTree.h [hunk 3]
```diff
@@ -177,7 +178,7 @@ class EntityTree : public Octree, public SpatialParentTree {
     static QByteArray remapActionDataIDs(QByteArray actionData, QHash<EntityItemID, EntityItemID>& map);
 
     QVector<EntityItemID> sendEntities(EntityEditPacketSender* packetSender, EntityTreePointer localTree,
-                                       float x, float y, float z);
+                                       QString entityHostType, float x, float y, float z);
```
Comment (HifiExperiments) [inline libraries/entities/src/EntityTree.h:181]: this one too!

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-f2204e8_5f0caf7d.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-f2204e8.tar_5f0caf7d.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-f2204e8.tar_5f0caf7d.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-f2204e8_5f0caf7d.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-f2204e8_5f0caf7d.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d.tar_859c78d8.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d.tar_859c78d8.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d_859c78d8.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d_859c78d8.exe

**Comment by 1st-BrainStormer** [comment]
Tested PR  Pasted entities to all 3 host types, all worked as expected.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d_859c78d8.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d.tar_859c78d8.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d.tar_859c78d8.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d_859c78d8.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-9f1cc3d_859c78d8.exe

File: interface/src/scripting/ClipboardScriptingInterface.h [hunk 4]
```diff
@@ -117,10 +117,11 @@ class ClipboardScriptingInterface : public QObject {
      * Pastes the contents of the clipboard into the domain.
      * @function Clipboard.pasteEntities
      * @param {Vec3} position -  The position to paste the clipboard contents at.
+     * @param {Entities.EntityHostType} [entityHostType="domain"] - The type of entity to create.
```
Comment (ctrlaltdavid) [inline interface/src/scripting/ClipboardScriptingInterface.h:120]:
```suggestion
     * @param {Entities.EntityHostType} [entityHostType="domain"] - The type of entities to create.
```

File: interface/src/scripting/ClipboardScriptingInterface.h [hunk 5]
```diff
@@ -117,10 +117,11 @@ class ClipboardScriptingInterface : public QObject {
...
      * @returns {Uuid[]} The IDs of the new entities that were created as a result of the paste operation. If entities couldn't 
      *     be created then an empty array is returned.
-     */
-    Q_INVOKABLE QVector<EntityItemID> pasteEntities(glm::vec3 position);
+    */
```
Comment (ctrlaltdavid) [inline interface/src/scripting/ClipboardScriptingInterface.h:123]:
```suggestion
     */
```

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8b89ec0_e1ab52d9.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8b89ec0.tar_e1ab52d9.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8b89ec0.tar_e1ab52d9.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8b89ec0_e1ab52d9.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR497/Vircadia-Alpha-PR497-8b89ec0_e1ab52d9.exe

**Pull request merged**